crates/flux-fhir-analysis/src/conv/mod.rs

```diff
@@ -558,11 +559,13 @@ impl<'genv, 'tcx: 'genv, P: ConvPhase<'genv, 'tcx>> ConvCtxt<P> {
         } else {
             self.conv_expr(&mut env, &variant.ret.idx)?
         };
+        let invariants = invariants.to_vec();
```

Comment:
We should apply the invariants here.

Reply:
Did you mean in `conv_struct_variant` (since we're just using the empty list for `enum`?)

Reply:
ah yeah, I put the comment on the wrong code :P